### PR TITLE
[changelog skip] Move integration logging test to main test suite

### PR DESCRIPTION
**Pull request by harrylewis**
### Description

This PR moves the first of 3 integration tests mentioned in #2172 that live in the `test/shell` directory into the main test suite.

There was a suggestion to remove the test altogether if there are current tests that exercise the functionality. A cursory look through the existing tests indicates to me that this test is fairly unique.

I decided to move the first test, the "does it log" test to the `TestIntegrationSingle`, as it seemed to me to fit best into this kind of basic configuration scenario (hopefully I am understand the context of these tests correctly).

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` the pull request title.
- [x] I have added appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

**Comment by harrylewis** [comment]
I feel the PR is still needs a bit of work. Since this is my first contribution to Puma, I wanted to get feedback in early. Not sure how to assign reviewers, or labels. I'm assuming that is done by the project maintainers.

File: test/test_integration_single.rb [hunk 1]
```diff
@@ -112,4 +112,23 @@ def test_siginfo_thread_print
 
     assert_match "Thread: TID", output.join
   end
+
+  def test_write_to_log
+    skip_unless_signal_exist? :TERM
+
+    suppress_output = '> /dev/null 2>&1'
```
Comment (harrylewis) [inline test/test_integration_single.rb:119]: I wanted to clean up the test output, so I suppressed the `curl` return.

File: test/test_integration_single.rb [hunk 2]
```diff
@@ -112,4 +112,23 @@ def test_siginfo_thread_print
 
     assert_match "Thread: TID", output.join
   end
+
+  def test_write_to_log
```
Comment (harrylewis) [inline test/test_integration_single.rb:116]: Most of this is copied from `test/shell/t1.rb`. One of the main modifications was to remove the `#sleep` calls. I felt they were unnecessary now given the helper methods `#cli_server`, and `#stop-server` wait until the Puma starts fully before handing control back to the test.

**Comment by nateberkopec** [comment]
Cheers! Good work.

**Pull request merged**